site-builder/src/publish.rs

```diff
@@ -26,22 +30,34 @@ use crate::{
     Config,
 };
 
-pub async fn publish_site(
-    directory: &Path,
-    content_encoding: &ContentEncoding,
-    site_name: &str,
-    config: &Config,
-    epochs: u64,
-    preprocess: bool,
-) -> Result<()> {
+#[derive(Parser, Debug)]
+pub struct PublishOptions {
+    /// The directory containing the site sources.
+    pub directory: PathBuf,
+    #[clap(short = 'e', long, value_enum, default_value_t = ContentEncoding::PlainText)]
+    /// The encoding for the contents of the site's resources.
+    pub content_encoding: ContentEncoding,
+    /// The name of the site.
+    #[clap(short, long, default_value = "test site")]
+    pub site_name: String,
+    /// The number of epochs for which to save the resources on Walrus.
+    #[clap(long, default_value_t = 1)]
+    pub epochs: u64,
+    /// Preprocess the directory before publishing.
+    /// See the `list-directory` command. Warning: Rewrites all `index.html` files.
+    #[clap(long, action)]
+    pub list_directory: bool,
+}
+
+pub async fn publish_site(publish_options: PublishOptions, config: &Config) -> Result<()> {
     edit_site(
-        directory,
-        content_encoding,
-        SiteIdentifier::NewSite(site_name.to_owned()),
+        &publish_options.directory,
+        &publish_options.content_encoding,
+        SiteIdentifier::NewSite(publish_options.site_name.to_owned()),
```

Comment:
Nit: The `to_owned` isn't necessary here.
```suggestion
        SiteIdentifier::NewSite(publish_options.site_name),
```